cmd/symbols/internal/api/handler_cgo.go

```diff
@@ -69,3 +96,33 @@ func (s *grpcService) SymbolInfo(ctx context.Context, request *proto.SymbolInfoR
 
 	return &response, nil
 }
+
+type localCodeIntelSender struct {
+	sendFunc func(symbols []*proto.LocalCodeIntelResponse_Symbol) error
+
+	buf []*proto.LocalCodeIntelResponse_Symbol
+}
+
+func newLocalCodeIntelSender(sendFunc func(symbols []*proto.LocalCodeIntelResponse_Symbol) error) *localCodeIntelSender {
+	return &localCodeIntelSender{
+		sendFunc: sendFunc,
+	}
+}
+
+func (s *localCodeIntelSender) Reset() {
+	if s.buf == nil {
+		return
+	}
+
+	s.buf = s.buf[:0]
```

Comment:
nit: slicing to zero is always safe, [even if the slice is nil](https://go.dev/play/p/CiG89YKD7wH)

```suggestion
	s.buf = s.buf[:0]
```

Reply:
TIL! It's not obvious to me though what in the go spec says that this should work. Maybe I need to sit down and finally watch https://www.youtube.com/watch?v=ynoY2xz-F8s